script.js

```diff
@@ -1 +1,519 @@
+    const currNode = cardHtml.childNodes[i];
+    if (currNode.className === 'displayName') {
+      currNode.innerHTML = card.display;
+    } else {
+      currNode.innerHTML = card.symbol;
+    }
+  }
+};
+
+// Fn to clear text by each nested div inside square divs
+const clearCardDisplay = (cardHtml) => {
+  if (cardHtml) {
+    for (let i = 0; i < cardHtml.childNodes.length; i += 1) {
+      const currNode = cardHtml.childNodes[i];
+      if (currNode.className === 'displayName') {
+        currNode.innerHTML = '';
+      } else {
+        currNode.innerHTML = '';
+      }
+    }
+  }
+};
+
+// Say congrats and disappear in 5 seconds
+const congratulations = (timeLeft) => {
+  const congratsDisplay = document.createElement('div');
+  console.log(timeLeft);
+  congratsDisplay.innerHTML = `<img src="https://media4.giphy.com/media/g9582DNuQppxC/200.gif"/>   Congratulations! You\'ve won the game in ${timeLeft} seconds!</img>`;
+  setTimeout(() => {
+    congratsDisplay.innerHTML = '';
+  }, 5000);
+  document.body.appendChild(congratsDisplay);
+};
+const startGameTimer = (time) => {
+  if (numOfClicks === 1) {
+    time = 180;
+    const initialMessage = 'Time left: ';
+    timer.innerHTML = initialMessage + time + 's';
+
+    ref = setInterval(() => {
+      time -= 1;
+      timeLeft = countDownTime - time;
+      timer.innerHTML = initialMessage + time + 's';
+    }, 1000);
+
+    // if game ends before all cards are matched, clear interval
+    setTimeout(() => {
+      clearInterval(ref);
+      let x = 3;
+      // display message to user that game is restarting in 3 seconds synchronously
+      const ref4 = setInterval(() => {
+        timer.innerHTML = `Time\'s up! You lost. Restarting in ${x}...`;
+        x -= 1;
+        if (x === 0) {
+          clearInterval(ref4);
+        }
+      }, 1000);
+      // restart the game after 3 seconds
+      setTimeout(() => {
+        noClickResetGame();
+      }, 4000);
+    }, 180000); // 180s or 3mins
+  } else if (numOfClicks > 1 && score === boardSize) {
+    clearInterval(ref);
+  }
+};
+const displayScore = () => `Your score is ${score}.`;
+
+const squareClick = (cardElement, column, row) => {
+  numOfClicks += 1;
+
+  // firstCard is the card object
+  // cardElement is HTML
+  if (firstCard === null && secondCard === null) {
+    firstCol = column;
+    firstRow = row;
+    currentNotMatched = true;
+    console.log(firstCardHtml, 'firstCardHtml');
+
+    if (positionArray[column][row] === 'matched') {
+      outputDivTag.innerHTML = 'Illegal, you cannot choose a matched card <br> Please click another card.';
+      // cannot clear the position of the matched card
+    } else {
+      firstCard = board[column][row];
+      console.log(firstCard, 'first card is reassigned');
+      positionArray[column][row] = 'x';
+      // turn this card over
+
+      // Tracking the firstCard's HTML in firstCardHTML
+      // clearCardDisplay(firstCardHtml);
+      // clearCardDisplay(secondCardHtml);
```

Comment:
Please remove unused code before submitting. This is like leaving rubbish around the house :p